### PR TITLE
docs: Rename Azure IPAM to Azure IPAM (Legacy)

### LINKED ISSUE #27289: Update Azure docs to the current
Creating this issue to update multiple azure docs to relevant based on this comment:

https://github.com/cilium/cilium/pull/27199#pullrequestreview-1561391819 

PR #27199 

```[tasklist]
### Tasks
- [ ] Add a new Azure CNI Powered by Cilium entry to External Installers and add delegated ipam section https://docs.cilium.io/en/latest/installation/external-toc/ @tamilmani1989
- [ ] Getting Started guide https://docs.cilium.io/en/latest/gettingstarted/k8s-install-default/. Rework the AKS tabs so that only instructions for BYOCNI remain. Completely remove Azure IPAM.  @tamilmani1989
- [ ] Helm guide https://docs.cilium.io/en/latest/installation/k8s-install-helm/. Update Getting started instructions for AKS @tamilmani1989
- [ ] IPAM guides https://docs.cilium.io/en/latest/network/concepts/ipam/. Rename Azure IPAM to Azure IPAM (Legacy)  @tamilmani1989
- [ ] Azure IPAM https://docs.cilium.io/en/latest/network/concepts/ipam/azure/. Rename  Azure IPAM page to Azure IPAM (Legacy) and reason out why its legacy @tamilmani1989
- [ ] CNI chaining guides https://docs.cilium.io/en/latest/installation/cni-chaining/. Rename Azure CNI to Azure CNI (Legacy). @vipul-21
- [ ] Azure CNI chaining https://docs.cilium.io/en/latest/installation/cni-chaining-azure-cni/.  @vipul-21
- [x] NodePort XDP on Azure https://docs.cilium.io/en/latest/network/kubernetes/kubeproxy-free/#nodeport-xdp-on-azure  @vipul-21
```

### PR DESCRIPTION
**Pull request by tamilmani1989**
Update Azure Ipam to Legacy and add recommended ways to install cilium on AKS

Fixes: #27289

### CODE REVIEW

File: Documentation/network/concepts/ipam/azure.rst [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
+###################
+Azure IPAM (Legacy)
+###################
 
 .. note::
 
-   While still maintained for now, Azure IPAM is considered legacy and is not
-   compatible with AKS clusters created in `Bring your own CNI <https://docs.microsoft.com/en-us/azure/aks/use-byo-cni?tabs=azure-cli>`_
-   mode. Using BYOCNI is the preferred way to install Cilium on AKS.
+   While still maintained for now, Azure IPAM is considered legacy. The recommended way to install cilium on AKS is
```
Comment (qmonnet) [inline Documentation/network/concepts/ipam/azure.rst]: Nit - There are two ways

```suggestion
   While still maintained for now, Azure IPAM is considered legacy.
   The recommended ways to install cilium on AKS are 
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good beside one minor suggestion

**Review by tommyp1ckles — APPROVED** [review]

**Review by qmonnet — COMMENTED** [review]
Thanks for the change, can you please squash the two commits together and drop the `Co-authored-by:` tag (or use my work email instead of personal one that GitHub autofilled)?

**Comment by nbusseneau** [comment]
I am switching PRs related to #27289 to draft so that I have a chance to review them cohesively since they're all out at the same time and are all supposed to work together.

**Comment by joestringer** [comment]
This PR seems to already be superseded by https://github.com/cilium/cilium/pull/28910 , should we close this one?

**Comment by tamilmani1989** [comment]
closing as similar PR exists https://github.com/cilium/cilium/pull/28910

**Pull request closed without merging**